kubernetes.go

```diff
@@ -321,6 +339,7 @@ func (c *KubernetesClient) Watch() error {
 		}
 
 		if !more {
+			c.stopWatchers()
```

Comment:
Good point, probably the first call to something like the connect method should be made here in watch (and not in the constructor) as at the end it the only stateful things it creates are the watchers.

I'll do another PR with some refactorizations about this.